app/assets/javascripts/discourse/app/components/user-menu/messages-list.js

```diff
@@ -51,9 +57,11 @@ export default class UserMenuMessagesList extends UserMenuNotificationsList {
     );
     const content = [];
 
-    const notifications = data.notifications.map((n) => Notification.create(n));
-    await Notification.applyTransformations(notifications);
-    notifications.forEach((notification) => {
+    const unreadNotifications = data.unread_notifications.map((n) =>
+      Notification.create(n)
+    );
+    await Notification.applyTransformations(unreadNotifications);
```

Comment:
What do you think about centralising this logic into a function called `initialiseNotifications`?